### PR TITLE
UI-Rework pull into master

### PR DESCRIPTION
**Pull request by alexdlukens**
Made landing page for idea shop portal the "welcome" page, added new "trainings" UI, "Shop Status" template tab, and link to Idea Shop Wiki. Moved Login/logout buttons to top navbar, and made it so a few pages are navigable without logging in.

### CODE REVIEW

File: safety-quiz/safetyQuiz.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 import requests
 from flask import current_app
 
-from checkIn.model import User, UserLocation, Type, Training, Machine, Quiz, Question, Option, MissedQuestion, init_db, Major, College, HawkCard
```
Comment (AlirpJ) [inline safety-quiz/safetyQuiz.py:21]: You don't need "checkIn." ?

File: safety-quiz/safetyQuiz.py [hunk 2]
```diff
@@ -80,9 +88,71 @@ def index():
...
+                  .filter_by(location_id=2)
+                  .all())]
+
+    if request.method =='GET':
+        return render_template('shop_status.html', user_count=user_count, in_lab=in_lab)
+    elif request.method == 'POST':
+        response = app.make_response('<h1>Not yet implemented!</h1>'), 418
+        return response
+
+@app.route('/COVID',methods=['GET', 'POST'])
```
Comment (AlirpJ) [inline safety-quiz/safetyQuiz.py:141]: This function and route has been moved outside of safetyQuiz.py

File: safety-quiz/safetyQuiz.py [hunk 3]
```diff
@@ -80,9 +87,71 @@ def index():
         quizzes = db.query(Machine).filter(Machine.quiz_id != None).order_by(Machine.quiz_id).all()
         return render_template('admin/index.html', trainings=trainings, quizzes=quizzes)
     else:
-        return render_template('index.html', trainings=trainings)
+        return render_template('welcome.html', trainings=trainings)
+
+@app.route('/new_trainings')
```
Comment (pkarnia) [inline safety-quiz/safetyQuiz.py:92]: move route to a blueprint, maybe userflow.py

File: safety-quiz/safetyQuiz.py [hunk 4]
```diff
@@ -80,9 +87,71 @@ def index():
...
+def new_training_interface():
+    db = db_session()
+
+    trainings = db.query(Training).outerjoin(Machine).filter(Training.trainee_id == session['sid']).filter(\
+    Training.invalidation_date == None).filter(Machine.location_id.in_((2, 3))).order_by(Training.in_person_date).all()
+    quizzes = db.query(Machine).filter(Machine.quiz_id != None).order_by(Machine.quiz_id).all()
+    training_count = len(trainings)
+    return render_template('new_trainings.html', trainings=trainings, quizzes=quizzes, training_count = training_count)
+
+@app.route('/welcome')
```
Comment (pkarnia) [inline safety-quiz/safetyQuiz.py:102]: move route to a blueprint, maybe public.py

File: safety-quiz/safetyQuiz.py [hunk 5]
```diff
@@ -80,9 +87,71 @@ def index():
...
+        .filter_by(timeOut=None) \
+        .filter_by(location_id=2)
+        .all())]
+
+
 
 
+    return render_template('welcome.html', user_count = user_count, in_lab = in_lab)
+
+@app.route('/shop_status', methods=['GET','POST'])
```
Comment (pkarnia) [inline safety-quiz/safetyQuiz.py:121]: move route to a blueprint, maybe public.py

File: safety-quiz/safetyQuiz.py (hunk 2 above)
Comment (pkarnia) [reply safety-quiz/safetyQuiz.py:140]: Still needs to be resolved

**Review by pkarnia — CHANGES_REQUESTED** [review]
You should also consider moving any Idea Shop JS to be served via flask templating not as a static to automatically set caching in the event of JS changes.

**Review by pkarnia — APPROVED** [review]

**Pull request merged**